dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -167,24 +185,32 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                     }
                     else
                     {
+                        auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+
                         // AppId is generated by PushNotificationLongRunningTask singleton
                         wil::unique_cotaskmem_string unpackagedAppUserModelId;
-                        RegisterUnpackagedApplicationHelper(remoteId, unpackagedAppUserModelId);
-                        THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(unpackagedAppUserModelId, remoteId, channelInfo));
+                        THROW_IF_FAILED(notificationPlatform->RegisterFullTrustApplication(m_processName.get(), remoteId, &unpackagedAppUserModelId));
 
-                        auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+                        THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(unpackagedAppUserModelId, remoteId, channelInfo));
 
-                        wil::unique_cotaskmem_string processName;
-                        THROW_IF_FAILED(GetCurrentProcessPath(processName));
-                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(processName.get(), GetComServerClsid()));
+                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(m_processName.get(), m_registeredClsid));
```

Comment:
Add a comment here to explain why we need to do this after channel creation. There is a legit reason which folks reading the code for the first time won't understand.

Reply:
Also can this be moved to RegisterForegroundSinkHelper?

Reply:
yes we can move this into RegisterForegroundSinkHelper